main.cpp

```diff
@@ -53,13 +55,53 @@ struct Wrapper
 {
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
-        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+        //std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl;
+        std::cout << "Wrapper(" << type_name<decltype(val)>() << ")" << std::endl;
     }
+
+    void print()    // #5
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+
+private:
+    Type val;
 };
 
-int main()
+template<>
+struct Wrapper<Point>
 {
-    variadicHelper( 3, std::string("burgers"), 2.5, Point{3.f, 0.14f} );
+    Wrapper(Point&& p) : point(std::move(p)) 
+    { 
+        //std::cout << "Wrapper(" << typeid(point).name() << ")" << std::endl; 
+        std::cout << "Wrapper(" << type_name<decltype(point)>() << ")" << std::endl;
+    }
+
+    void print()    // #5
+    {
+        std::cout << "Wrapper::print(" << point.toString() << ")" << std::endl;
+    }
+
+private:
+    Point point;
+};
+
+// #3
+template<typename T, typename ... Args>
+void variadicHelper(T first, Args&& ... args)
```

Comment:
Your first param should be an R-value reference.  The current form is forcing a copy.  
- [ ] revise

Reply:
Done.